leshan-server-core/src/main/java/org/eclipse/leshan/server/queue/PresenceServiceImpl.java

```diff
@@ -67,7 +72,7 @@ public boolean isClientAwake(Registration registration) {
      */
     public void setAwake(Registration reg) {
         if (reg.usesQueueMode()) {
-            PresenceStatus status = new PresenceStatus();
+            PresenceStatus status = new PresenceStatus(awakeTimeInfo.usedClientAwakeTime(reg));

```

Comment:
I'm even asking myself if we can not consider that a device is `awake` when it is in the `clientStatusList` of `PresenceServiceImpl` and `sleeping` when it is not.

If we do that, I feel we don't need `PresenceStatus` anymore as the only remain attribute is `clientScheduledFuture` so we can put it directly in the `clientStatusList` map.

Reply:
Yes, it's true that `clientAwakeTime` is not needed inside `PresenceStatus` anymore, since it is directly taken with the Awake Time Provider by `PresenceService`. 

> I'm even asking myself if we can not consider that a device is awake when it is in the clientStatusList of PresenceServiceImpl and sleeping when it is not.

I think that is simplifying it too much and making the code more unclear... `setSleeping` will be to remove a `ScheduledFuture` from a Map, and `setAwake` to add it... Doesn't sound very intuitive FMPOV. The enumerate with `Awake` and `Sleeping` states makes it more clear I think. Also in my original implementation, the `PresenceStatus` was a way to handle every client's queue mode independently. For example, the queues to store requests are also inside this object... We can do it your way but I think is more clear and can be more useful in a future to have this `PresenceStatus` object. Otherwise adding more features could be difficult, and of course we can also remove it in the future if we still see that there is no point in having this object. 

What do you think?

Reply:
>  For example, the queues to store requests are also inside this object...

FMPOV, If we implements this "queue of requests", it  should be handled by another object which just use the PresenceService and Send request. I mean this is on top of what we are implementing not inside.

I also feel this could simplify thread-safeness, but let's this aside. I will maybe try it on my side and propose a PR to talk about this specifically. Not critical this is just internal implementation.

Reply:
Ok, I also don't think is a critical aspect right now, we can discuss it in another PR.